plugins/bufferlines/barbar.nix

```diff
@@ -69,249 +24,592 @@ let
+                enabled = helpers.defaultNullOpts.mkBool false ''
+                  Enable showing diagnostics of this `|diagnostic-severity|` in the 'tabline'.
+                '';
+
+                icon = helpers.defaultNullOpts.mkStr iconDefault ''
+                  The icon which accompanies the number of diagnostics.
+                '';
+              };
+            in
+            {
+              "vim.diagnostic.severity.ERROR" = mkDiagnosticIconOptions " ";
+              "vim.diagnostic.severity.HINT" = mkDiagnosticIconOptions "󰌶 ";
+              "vim.diagnostic.severity.INFO" = mkDiagnosticIconOptions " ";
+              "vim.diagnostic.severity.WARN" = mkDiagnosticIconOptions " ";
+            };
+        };
+        apply = helpers.toRawKeys;
+        default = { };
+        description =
+          helpers.defaultNullOpts.mkDesc
+            {
+              "vim.diagnostic.severity.ERROR" = {
+                enabled = false;
+                icon = " ";
+              };
+              "vim.diagnostic.severity.HINT" = {
+                enabled = false;
+                icon = "󰌶 ";
+              };
+              "vim.diagnostic.severity.INFO" = {
+                enabled = false;
+                icon = " ";
+              };
+              "vim.diagnostic.severity.WARN" = {
+                enabled = false;
+                icon = " ";
+              };
+            }
+            ''
+              Set the icon for each diagnostic level.
+              The keys will be automatically translated to raw lua:
+              ```nix
+                {
+                  "vim.diagnostic.severity.INFO".enabled = true;
+                  "vim.diagnostic.severity.WARN".enabled = true;
+                }
+              ```
+              will result in the following lua:
+              ```lua
+                {
+                  -- Note the table keys are not string literals:
+                  [vim.diagnostic.severity.INFO] = { ['enabled'] = true },
+                  [vim.diagnostic.severity.WARN] = { ['enabled'] = true },
+                }
+              ```
+            '';
+      };
+
+      gitsigns =
+        helpers.defaultNullOpts.mkAttrsOf
+          (
+            with types;
+            submodule {
+              freeformType = with types; attrsOf anything;
+              options = {
+                enabled = mkOption {
+                  type = types.bool;
+                  description = helpers.defaultNullOpts.mkDesc true ''
+                    Enables showing git changes of this type.
+                    Requires `|gitsigns.nvim|`.
+                  '';
+                };
 
-    maximumLength = helpers.defaultNullOpts.mkInt 30 "Sets the maximum buffer name length.";
+                icon = mkOption {
+                  type = types.str;
+                  example = "";
+                  description = ''
+                    The icon which accompanies the number of git status.
```

Comment:
Using `mkOption` here to avoid setting `Plugin default`? If so could you add a comment?

Should the type not be `nullOr (maybeRaw str)`? If so, use `helpers.mkNullOrStr`.

Reply:
I didn't want it to be nullable, but now I'm not sure...